metrics.go

```diff
@@ -120,23 +124,45 @@ func (m *Metrics) MeasureSinceWithLabels(key []string, start time.Time, labels [
 			key = insert(0, m.ServiceName, key)
 		}
 	}
-	if !m.allowMetric(key) {
+	allowed, labelsFiltered := m.allowMetric(key, labels)
+	if !allowed {
 		return
 	}
 	now := time.Now()
 	elapsed := now.Sub(start)
 	msec := float32(elapsed.Nanoseconds()) / float32(m.TimerGranularity)
-	m.sink.AddSampleWithLabels(key, msec, labels)
+	m.sink.AddSampleWithLabels(key, msec, labelsFiltered)
 }
 
 // UpdateFilter overwrites the existing filter with the given rules.
 func (m *Metrics) UpdateFilter(allow, block []string) {
+	m.UpdateFilterAndLabels(allow, block, m.AllowedLabels, m.BlockedLabels)
+}
+
+// UpdateFilterAndLabels overwrites the existing filter with the given rules.
```

Comment:
Can we make this doc comment explicit what the behaviour of label filter is? i.g. apply globally to all metrics and apply orthogonally to the allow/block prefixes. I'd also include the use-case for them (globally configuring whether potentially high-cardinality metrics are included) to make it obvious why they work like this.

I think that satisfied the last point Kyle mentioned.

Reply:
@banks DONE: I added more text in README.md as well